src/Microsoft.OData.Core/WriterValidationUtils.cs

```diff
@@ -32,11 +38,27 @@ internal static void ValidateMessageWriterSettings(ODataMessageWriterSettings me
 
             if (messageWriterSettings.BaseUri != null && !messageWriterSettings.BaseUri.IsAbsoluteUri)
             {
+#if NETCOREAPP3_1_OR_GREATER || NETSTANDARD2_0_OR_GREATER
+                using (Activity activity = source.StartActivity("Validation Error"))
+                {
+                    activity?.SetTag("Validation Type", "Writer Validation");
+                    activity?.SetTag("Message", Strings.WriterValidationUtils_MessageWriterSettingsBaseUriMustBeNullOrAbsolute(UriUtils.UriToString(messageWriterSettings.BaseUri)));
+                }

```

Comment:
My 2 cents: the validation error should be an event rather than an activity, because we are tracing an operation here but rather logging that something has happened. And these events can be correlated to an activity.

We should probably use constants for the activity, event and tag names. But we can look at that after the hackthon.